### PR TITLE
Add GCS artifact generation to all jobs

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

/cc @kargakis 

test job https://ci.openshift.redhat.com/jenkins/job/skuznets/2/console

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
https://ci.openshift.redhat.com/jenkins/job/skuznets/3/console

**Comment by stevekuznetsov** [comment]
Artifacts being published here: https://console.cloud.google.com/storage/browser/origin-ci-test/pr-logs/pull/15149/?project=openshift-gce-devel

File: sjb/gcs/finished.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    started_data = load(started_file)
+
+commit = check_output(
+    ["git", "log", "-1", "--pretty=%H"],
+    cwd="'/data/src/github.com/openshift/origin"
+)
+
+metadata = load(urlopen("{}api/json".format(getenv("BUILD_URL"))))
+
+with open("/tmp/finished.json", "w+") as finished_file:
```
Comment (kargakis) [inline sjb/gcs/finished.py]: Comment that we are following the upstream format.

File: sjb/gcs/started.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    pull_id = getenv(repository_env_name + "_PULL_ID", "")
+    if len(pull_id) > 0:
+        pull_sha = check_output(
+            ["git", "log", "-1", "--pretty=%H", "pull-{}".format(pull_id)],
+            cwd=repository_dir
+        ).strip()
+        version_info += ",{}:{}".format(pull_id, pull_sha)
+
+    if repository == "origin":
+        # today we only support PULL_REFS for origin
```
Comment (kargakis) [inline sjb/gcs/started.py]: Maybe "today we support PULL_REFS only for origin"?

File: sjb/gcs/started.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+version_script_path = "/tmp/origin_version.sh"
+with open(version_script_path, "w+") as version_script_file:
+    version_script_file.write(version_script)
+
+version = check_output(
+    ["bash", version_script_path],
+    cwd=join(repository_base_dir, "origin")
+)
+
+with open("/tmp/started.json", "w+") as started_file:
```
Comment (kargakis) [inline sjb/gcs/started.py]: Comment that we are following the upstream format.

**Comment by kargakis** [comment]
Nice! I can see only started.json inside the artifacts ...

**Comment by kargakis** [comment]
```
+ sjb/gcs/finished.py
Traceback (most recent call last):
  File "sjb/gcs/finished.py", line 11, in <module>
    with open("/tmp/started.json", "r") as started_file:
IOError: [Errno 2] No such file or directory: '/tmp/started.json'
++ export status=FAILURE
++ status=FAILURE
+ set +o xtrace
########## FINISHED STAGE: FAILURE: RECORD THE ENDING METADATA [00h 00m 01s] ##########
```

**Comment by stevekuznetsov** [comment]
Yeah, didn't have time to investigate that last night. Will look at why the starting JSON is missing

**Comment by stevekuznetsov** [comment]
Artifacts look good: https://console.cloud.google.com/storage/browser/origin-ci-test/pr-logs/pull/15149/skuznets/14/?project=openshift-gce-devel

**Comment by kargakis** [comment]
👍

**Pull request merged**